api/base/exceptions.py

```diff
@@ -3,16 +3,39 @@
 from rest_framework import status
 from rest_framework.exceptions import APIException, ParseError
 
+def dict_error_formatting(errors, error):
```

Comment:
It appears that the first argument (`errors`) is only passed in as a reference to an external object that is modified in-place. (`errors` is defined outside the function and we simply append to it here)

In general, strongly prefer returning a value from the function instead, and appending that return value to `errors` as appropriate- it would make the code much clearer (and a lot less fragile).

For example, this function could return a `formatted_error_list`, and line 52 would change to `errors.extend(dict_error_formatting(message))`